### PR TITLE
Feature/set cookie defaults

### PR DESCRIPTION
**Pull request by cjhowald**
Note: This branch includes PR #26 changes. #26 should be merged first.

This fix sets cookie options consistently on all put and remove calls. It no longer relies on defaults being set in solution center (those should now be removed). The fix is done is such a way that these defaults will not affect users of our auth service, so if they use $cookies themselves for other purposes those cookies will be unaffected.

Also includes some code style improvements and cleanup of the tests.

### CODE REVIEW

File: src/authentication-service.js [hunk 1]
```diff
@@ -75,13 +57,12 @@ angular.module('sc-authentication', ['ngStorage', 'ngCookies', 'angular-jwt'])
...
          */
-        $get: [
-          '$q', '$localStorage', '$cookies', 'environmentsService', '$window', '$injector', 'jwtHelper', '$location', '$timeout',
-          authenticationFactory
-        ]
+        $get: authenticationFactory
       };
     }]);
 
+authenticationFactory.$inject = ['$q', '$localStorage', '$cookies', 'environmentsService', '$window', '$injector', 'jwtHelper', '$location', '$timeout'];
```
Comment (gregdeane) [inline src/authentication-service.js]: Nice update. Style guide rules!

File: src/authentication-service.js [hunk 2]
```diff
@@ -443,4 +424,18 @@ function authenticationFactory($q, $localStorage, $cookies, environmentsService,
...
+   */
+  function getCookieConfig() {
+    var env = self.getEnvironment();
+    var now = new Date();
+    return {
+      domain: env.DOMAIN,
+      secure: env.NAME !== 'LOCAL',
+      expires: new Date(now.getFullYear(), now.getMonth(), now.getDate() + 7)
+    };
+  }
```
Comment (gregdeane) [inline src/authentication-service.js]: Very clean.

File: test/authentication-service.spec.js [hunk 3]
```diff
@@ -40,32 +40,8 @@ describe('authenticationService', function () {
...
-            jwtHelper = _jwtHelper_;
-            $window = _$window_;
-
-            spyOn(environmentsService, 'getDomain').and.returnValue(mockedDomain);
-            spyOn(environmentsService, 'getTokensAPI').and.returnValue(mockedTokensAPIEndpoint);
-            spyOn(jwtHelper, 'decodeToken').and.returnValue(mockedTokenPayload);
-            spyOn(authenticationService, 'setToken').and.callThrough();
-          });
+      modules();
+      injectors();
```
Comment (gregdeane) [inline test/authentication-service.spec.js]: Thanks for updating. Much cleaner now.

**Comment by gregdeane** [comment]
Tests pass, manual testing looks good.

**Comment by gregdeane** [comment]
👍

**Comment by gregdeane** [comment]
👍

**Comment by dami-gg** [comment]
:+1:

**Pull request merged**